bolt-design-system/bolt
# Use pseudo patterns to reduce redundant twig markup

## Jira

http://vjira2:8080/browse/BDS-767 (sort of).  See discussion on #958

## Summary

Removes redundant markup in press and media (WIP)

## Details

See discussion on #958 for background

This PR uses [pseudo patterns](https://patternlab.io/docs/pattern-pseudo-patterns.html) to accomplish the following two things at once:
- Avoid redundant twig templates
- Show the correct source twig in Pattern Lab's "pattern info" feature

This technique could also be used for other sections including:
- Agenda manager detail page examples
- Blog posts page examples

One drawback to the way I'm doing it in this PR is that it creates an additional default page at `/pattern-lab/?p=pages-d8-browse`.  One way around that would be to turn the default page into one of the two examples using a [markdown file](https://patternlab.io/docs/pattern-documenting.html) to properly set the pattern name.

## How to test

- Go to `/pattern-lab/?p=pages-d8-browse-press` and `/pattern-lab/?p=pages-d8-browse-media`
- Click 'Show Pattern Info' in the PL UI and confirm that both pages show the full source twig
